src/base58.cpp

```diff
@@ -323,67 +323,60 @@ bool CBitcoinSecret::SetString(const std::string& strSecret)
     return SetString(strSecret.c_str());
 }
 
-bool CZCPaymentAddress::Set(const libzcash::PaymentAddress& addr)
+template<class DATA_TYPE, CChainParams::Base58Type PREFIX, size_t SER_SIZE>
+bool CZCEncoding<DATA_TYPE, PREFIX, SER_SIZE>::Set(const DATA_TYPE& addr)
 {
     CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
     ss << addr;
     std::vector<unsigned char> addrSerialized(ss.begin(), ss.end());
-    assert(addrSerialized.size() == libzcash::SerializedPaymentAddressSize);
-    SetData(Params().Base58Prefix(CChainParams::ZCPAYMENT_ADDRRESS), &addrSerialized[0], libzcash::SerializedPaymentAddressSize);
+    assert(addrSerialized.size() == SER_SIZE);
```

Comment:
If size is not as expected, return false, instead of asserting.  Otherwise why have the function return a bool, when void would suffice.

Reply:
Good idea (though I don't think it's as simple as that, because at a cursory glance it appears that the returned `bool` is ignored), but out of scope for this PR, as I am just maintaining existing behaviour in this refactor. Please open an issue for it.